aggregate-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java

```diff
@@ -0,0 +1,51 @@
+package com.c12e.cortex.examples.aggregate;
+
+import com.c12e.cortex.examples.local.SessionExample;
+import com.c12e.cortex.phoenix.AggregationResult;
+import com.c12e.cortex.profiles.CortexSession;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+import com.c12e.cortex.phoenix.ProfileWindowedAggregation;
+import static com.c12e.cortex.examples.aggregate.Metrics.setKPI;
+
+
+/**
+ * Sample CLI application that writes the underlying Profile data to an Redis.
+ * Caches the Profile on profile_id
+ */
+@Command(name = "kpi-query", description = "Calculating KPI using aggregate using from Profiles", mixinStandardHelpOptions = true)
+public class KPIQueries implements Runnable {
+    @Option(names = {"-p", "--project"}, description = "Cortex Project to use", required = true)
+    String project;
+
+    @Option(names = {"-ps", "--profile"}, description = "Profile Schema Name", required = true)
+    String profileSchemaName;
+
+    @Override
+    public void run() {
+        var sessionExample = new SessionExample();
+        CortexSession cortexSession = sessionExample.getCortexSession();
+        runKPI(cortexSession, project);
+    }
+
+    public boolean runKPI(CortexSession cortexSession, String project){
+        Dataset<Row> profileData = cortexSession.read().profile(project, profileSchemaName).load().toDF();
+        profileData = profileData.sample(0.002);
+
+        String script = "filter(MAIL_STATUS.equalTo(\"MAIL CONTINUE\")).count()";
+//        profileData.filter(col("MAIL_STATUS").isin(new String[]{"MAIL CONTINUE"})).count()
+//        String script = "count()";
+        ProfileWindowedAggregation engine = new ProfileWindowedAggregation(profileData, "180 days");
+
+        AggregationResult result = engine.runWindowedAggregation(script);
+        Long firstRow = (Long) result.getDf().collectAsList().get(0).get(1);
+        System.out.println("========");
+        System.out.println(firstRow);
+        System.out.println("========");
```

Comment:
ensure there is a dataset named "cohort" associated with this, you can find it being set on the engine in CampaignService.runAgg

Reply:
agreed, given that this is being run on a single profile I did not see a reason to set it, also it seems from the ProfileScriptEngine that we can add as many datasets as we want and name them whatever(including "cohort"), and use them in the script. Should we keep it CampaignService agnostic or do we want to follow 1 cohort + 1 supplementary data format (as used in campaign services)

Reply:
I think the initial example should follow the same logic as what we have in phoenix to ensure the same results against the same expressions. If endusers want to further customize that is on them.

Reply:
```
Usage: profiles-example kpi-query [-hV] -d=<windowDuration> [-ed=<endDate>]
                                  -p=<project> -ps=<profileSchemaName>
                                  -s=<script> [-sd=<startDate>]
                                  [-cf=<cohortFilters>]...
Calculating KPI using aggregate using from Profiles
   -cf, --cohortFilters=<cohortFilters>
                            Cohort Filter
  -d, --duration=<windowDuration>
                            Window Duration
  -ed, --endDate=<endDate>
                            End Date, Set the time-frame over which the KPI is
                              calculated
  -h, --help                Show this help message and exit.
  -p, --project=<project>   Cortex Project to use
  -ps, --profile=<profileSchemaName>
                            Profile Schema Name
  -s, --script=<script>     KPI Script
  -sd, --startDate=<startDate>
                            Start Date, Set the time-frame over which the KPI
                              is calculated
  -V, --version             Print version information and exit.
```

with the latest commit, scripts like below work:
```
kpi-query -p local -ps cvs -s "filter(MAIL_STATUS.equalTo(\"MAIL CONTINUE\")).count()" -d "180 days"
kpi-query -p local -ps cvs  -d "180 days" -cf "MAIL_STATUS.equalTo(\"MAIL CONTINUE\")" -s "var profileCount = count(); var cohortCount = cohort.count(); cohortCount.divide(profileCount)" 
kpi-query -p local -ps cvs  -d "180 days" -cf "MAIL_STATUS.equalTo(\"MAIL CONTINUE\")" -cf "PEER_DESC.equalTo(\"MAPD\")" -cf "ELIGIBILITY.equalTo(\"ELIGIBLE\")" -s "var profileCount = count(); var cohortCount = cohort.count(); cohortCount.divide(profileCount)"
```